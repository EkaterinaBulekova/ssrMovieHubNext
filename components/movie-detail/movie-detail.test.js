import React from 'react';
import { shallow } from 'enzyme';
import MovieDetail from './movie-detail';
import toJson from 'enzyme-to-json';

describe('<MovieDetail />', () => {
  it('should render movie card with right data, class and onClick method', () => {
    const testMovie = {
        title: "Test movie",
        poster_path: "http://test-path",
        release_date:"2019-01-05",
        vote_average: 3.5,
        runtime: 120
    };
    const testClass=".movie-detail"
    const component = shallow(<MovieDetail/>);
    expect(toJson(component)).toMatchSnapshot();
    // expect(component.find(testClass+'-title').text()).toBe(testMovie.title);
    // expect(component.find(testClass+'-rating').text()).toBe(testMovie.vote_average.toString());
    // expect(component.find(testClass+'-duration').text()).toContain(testMovie.runtime.toString());
    // expect(component.find(testClass+'-release-date').text()).toBe((new Date(testMovie.release_date)).getFullYear().toString());
  });

});