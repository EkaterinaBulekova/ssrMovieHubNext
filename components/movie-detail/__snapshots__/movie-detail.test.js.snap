// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetail /> should render movie card with right data, class and onClick method 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
